packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,526 @@
+  /// included in the list of matches. Priority is given to matches that occur
+  /// on the edge of a render object's bounding box and to matches found by
+  /// [RenderBox.hitTest].
+  List<RenderObject> hitTest(Offset position, RenderObject root) {
+    final LinkedHashMap<RenderObject, bool> result = new LinkedHashMap<RenderObject, bool>();
+    final LinkedHashSet<RenderObject> edgeHits = new LinkedHashSet<RenderObject>();
+
+    _hitTestHelper(result, edgeHits, position, root, root.getTransformTo(null));
+    final Set<RenderObject> hits = new LinkedHashSet<RenderObject>();
+    hits.addAll(edgeHits);
+    result.forEach((RenderObject o, bool value) {
+      if (value)
+        hits.add(o);
+    });
+    return hits.toList();
+  }
+
+  void _inspectAt(Offset position) {
+    if (!_isSelectMode)
+      return;
+
+    final RenderIgnorePointer ignorePointer = _ignorePointerKey.currentContext.findRenderObject();
+    final RenderObject userRender = ignorePointer.child;
+    final List<RenderObject> selected = hitTest(position, userRender);
+
+    setState(() {
+      _selection.candidates = selected;
+    });
+  }
+
+  void _handlePanDown(DragDownDetails event) {
+    _lastPointerLocation = event.globalPosition;
+    _inspectAt(event.globalPosition);
+  }
+
+  void _handlePanUpdate(DragUpdateDetails event) {
+    _lastPointerLocation = event.globalPosition;
+    _inspectAt(event.globalPosition);
+  }
+
+  void _handlePanEnd(DragEndDetails details) {
+    // If the pan ends on the edge of the window assume that it indicates the
+    // pointer is being dragged off the edge of the display not a regular touch
+    // on the edge of the display. If the pointer is being dragged off the edge
+    // of the display we do not want to select anything. A user can still select
+    // a widget that is only at the exact screen margin by taping.
+    final Rect bounds = (Offset.zero & (ui.window.physicalSize / ui.window.devicePixelRatio)).deflate(_kOffScreenMargin);
+    if (!bounds.contains(_lastPointerLocation)) {
+      setState(() {
+        _selection.clear();
+      });
+    }
+  }
+
+  void _handleTap() {
+    if (!_isSelectMode)
+      return;
+    if (_lastPointerLocation != null) {
+      _inspectAt(_lastPointerLocation);
+
+      if (_selection != null) {
+        // Notify debuggers to open an inspector on the object.
+        developer.inspect(_selection.current);
+        print(_selection.current.toStringDeep());
+      }
+    }
+    setState(() {
+      _isSelectMode = false;
+    });
+  }
+
+  void _handleEnableSelect() {
+    setState(() {
+      _isSelectMode = true;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+    children.add(new GestureDetector(
+      onTap: _handleTap,
+      onPanDown: _handlePanDown,
+      onPanEnd: _handlePanEnd,
+      onPanUpdate: _handlePanUpdate,
+      behavior: HitTestBehavior.opaque,
+      excludeFromSemantics: true,
```

Comment:
If we wanted to make it accessible, we should add inspector functionality to the semantics debugger itself as the semantics debugger actually makes it in some ways easier to select widgets as all objects have unique bounding boxes, etc.